### PR TITLE
Add Apache License

### LINKED ISSUE #7: vscode-swift requires a LICENSE file
This will have the same issues as the other swift-server repositories https://github.com/swift-server/async-http-client/issues/523

### PR DESCRIPTION
**Pull request by 0xTim**
Resolves #7

### CODE REVIEW

**Comment by 0xTim** [comment]
@adam-fowler @svanimpe you'll need to explicitly approve this as existing contributors as we're effectively relicensing it.

@swift-server/sswg thoughts on copyright notice etc?

File: LICENSE
Comment (adam-fowler) [inline LICENSE:189]: @0xTim did you not just point out this issue in the AHC repo

Comment (0xTim) [reply LICENSE:189]: Yes, added this in case https://github.com/swift-server/async-http-client/issues/523#issuecomment-987971916 is right. Waiting for confirmation, but I won't merge until it's been OKed

File: scripts/generate_contributors_list.sh
Comment (svanimpe) [inline scripts/generate_contributors_list.sh:1]: Is this a safe assumption? I've had some issues with this in the past, so I now use `/usr/bin/env bash`

**Review by adam-fowler — APPROVED** [review]
This is good to go when copyright holder is updated

File: LICENSE
Comment (ktoso) [reply LICENSE:189]: This is fine as is. More details: https://github.com/swift-server/async-http-client/issues/523#issuecomment-987971916

File: scripts/generate_contributors_list.sh
Comment (adam-fowler) [reply scripts/generate_contributors_list.sh:1]: I think this is fine as is. I'm guessing this is a direct copy of the swift-nio contributors code. It'll only ever be run by the maintainers of the project.

**Review by svanimpe — APPROVED** [review]

**Comment by 0xTim** [comment]
Cool think we're good, just waiting for final input from the SSWG

**Comment by adam-fowler** [comment]
LGTM

**Pull request merged**
